iceberg/openhouse/internalcatalog/src/main/java/com/linkedin/openhouse/internal/catalog/fileio/FileIOConfig.java

```diff
@@ -62,4 +70,67 @@ FileIO provideLocalFileIO() {
       return null;
     }
   }
+
+  @Bean("S3FileIO")
+  S3FileIO provideS3FileIO() {
+    try {
+      S3Client s3 =
+          (S3Client) storageManager.getStorage(StorageType.S3).getClient().getNativeClient();
+      return new S3FileIO(() -> s3) {
+
+        @Override
+        public OutputFile newOutputFile(String path) {
+          return new DelegatingOutputFile(super.newOutputFile(PREFIX + path));
+        }
+
+        @Override
+        public InputFile newInputFile(String path, long length) {
+          return new DelegatingInputFile(super.newInputFile(PREFIX + path, length));
+        }
+
+        @Override
+        public InputFile newInputFile(String path) {
+          return new DelegatingInputFile(super.newInputFile(PREFIX + path));
+        }
+      };
+    } catch (IllegalArgumentException e) {
+      // If the S3 storage type is not configured, return null
+      // Spring doesn't define the bean if the return value is null
+      log.debug("S3 storage type is not configured", e);
+      return null;
+    }
+  }
+
+  @AllArgsConstructor
+  public static class DelegatingInputFile implements InputFile {
```

Comment:
seem hacky, is there a better way to solve this? should OH get rid of non-schemed paths completely?

Reply:
OpenHouseS3FileIO with delegation to S3FileIO makes sense to me even for implementing the future token or signed url based authentication.